persist/fs/commitlog/reader.go

```diff
@@ -109,58 +165,171 @@ func (r *reader) Read() (
 	annotation ts.Annotation,
 	resultErr error,
 ) {
-	entry, err := r.readEntry()
-	if err != nil {
-		resultErr = err
-		return
+	if r.nextIndex == 0 {
+		go r.readLoop()
+		for i, decoderBuf := range r.decoderBufs {
+			go r.decoderLoop(decoderBuf, r.outBufs[i])
```

Comment:
More of a style nit: could you capture `decoderBuf` and `r.outBufs[i]` in separate variables above this line, so that readers don't have to remember the closure argument evaluation semantics

Reply:
sure